flake8_plugins/assert_expected_actual.py

```diff
@@ -0,0 +1,113 @@
+import ast
+import json
+from pathlib import Path
+import re
+
+
+MSG = 'PTA100 expected should come before actual'
+
+
+def is_literal(node):
+    return (
+        isinstance(node, ast.Constant) or
+        (
+            (
+                isinstance(node, ast.List) or
+                isinstance(node, ast.Set) or
+                isinstance(node, ast.Tuple)
+            ) and
+            all(is_literal(elt) for elt in node.elts)
+        ) or
+        (
+            isinstance(node, ast.Dict) and
+            all(is_literal(key) for key in node.keys) and
+            all(is_literal(value) for value in node.values)
+        )
+    )
+
+
+def classify_arg(node):
+    info = {'type': type(node).__name__, 'literal': is_literal(node)}
+    if info['type'] == 'Name':
+        for keyword in ['actual', 'expected']:
+            if re.search(keyword, node.id, re.IGNORECASE):
+                info[keyword] = True
+    return info
+
+
+def classify_call(node):
+    info = {'args': len(node.args)}
+    if info['args'] == 2:
+        left, right = map(classify_arg, node.args)
+        info['left'] = left
+        info['right'] = right
+        if (
+            left.get('expected') or
+            right.get('actual') or
+            left.get('literal')
+        ):
+            info['good'] = True
+        if (
+            not info.get('good') and  # ignore cases with two literals
+            (
+                left.get('actual') or
```

Comment:
To make these easier to read it might actually be beneficial to store these as variables and use them as variables.

i.e.

```python
is_good = any([left.get('expected'), right.get('actual'), left.get('literal')])
is_bad = any([left.get('actual'), right.get('expected'), right.get('literal')])

if is_good:
    info['good'] = True
elif is_bad:
    info['flipped'] = True
```